src/main/java/info/kgeorgiy/java/advanced/student/StudentDB.java

```diff
@@ -0,0 +1,169 @@
+package info.kgeorgiy.java.advanced.student;
+
+import java.util.*;
+import java.util.function.BinaryOperator;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class StudentDB implements StudentQuery {
+
+    private static final String DEFAULT_EMPTY_STRING = "";
+
+    private enum Compare {
+        BY_NAME
+    }
+
+    private static final Map<Compare, Comparator<Student>> comparators = new HashMap<>() {{
```

Comment:
I didn't get the reason to has such map. You can just create private final Comporator<Student> COMPARATOR_BY_NAME

Reply:
I thought that Ican add something else here + extensibility, but yeah, in the end - no need for the map